reinforcement_learning/agents/common/lazy_tabular_action_value.py

```diff
@@ -19,62 +18,43 @@ class LazyTabularActionValue(BaseActionValue):
     MAX_PEN_WIDTH = 4
 
     def __init__(self):
-        self.action_value_dict = dict()
+        self.action_value_dict = defaultdict(lambda: defaultdict(lambda: self._default_cell_value))
 
     def __getitem__(self, key: tuple):
         assert len(key) == 2, f"Invalid key: {key}, should be tuple(BaseState, BaseAction)..."
 
         state, action = key
-
-        # Lazy initialization
-        if not self.action_value_dict.get(state):
-            self.action_value_dict[state] = {action: self._initial_cell_value}
-        elif not self.action_value_dict[state].get(action):
-            self.action_value_dict[state][action] = self._initial_cell_value
-
         return float(self.action_value_dict[state][action])
 
     def __setitem__(self, key, value):
         assert len(key) == 2, f"Invalid key: {key}, should be tuple(BaseState, BaseAction)..."
 
         state, action = key
+        self.action_value_dict[state][action] = float(value)
 
-        if not self.action_value_dict.get(state):
-            self.action_value_dict[state] = dict()
+    def max(self, state):
+        expected_returns = self.action_value_dict[state].values()
+        return max(expected_returns) if expected_returns else self._default_cell_value
 
-        self.action_value_dict[state][action] = float(value)
+    def argmax(self, state):
+        actions = self.action_value_dict[state]
+        return {key for (key, value) in actions.items() if value == max(actions.values())}
```

Comment:
What if there is no such a state?

Reply:
Well then action.items is empty so empty set is returned